src/dirs.proj

```diff
@@ -7,6 +7,8 @@
     <Project Include="Utilities\Microsoft.Build.Utilities.csproj" />
     <Project Include="XMakeBuildEngine\Microsoft.Build.csproj" />
     <Project Include="XMakeCommandLine\MSBuild.csproj" />
+    <Project Include="XMakeCommandLine\MSBuildTaskHost\MSBuildTaskHost.csproj"
+             Condition="'$(FullFrameworkBuild)' == 'true'" />
```

Comment:
Seems fine for now. The only thing I worry about is the `CLR2COMPATIBILITY` const to build some of the sources in that mode. I hope we didn't diverge too far and it becomes a huge pain to get things to build for 3.5. And it would be good if we refactor for that to break sooner than later.